### PR TITLE
Add include_otter tag for markdown cells in solution

### LINKED ISSUE #730: "Include" tag for markdown cells in question solution block
**Is your feature request related to a problem? Please describe.**
I would like to be able to put Markdown cells within a question, between two pieces of student code. i.e. in the solution block
e.g.
- A cell with a student written function
- An instructor written markdown cell to tell the student what to do in the next cell
- A cell where the student writes some more code using the function they wrote

**Describe the solution you'd like**
A way to tag Markdown cells within a solution block as "included" in student notebook

**Describe alternatives you've considered**

**Additional context**
Chris's context for why markdown cells are removed:
"otter’s logic for removing solution cells doesn’t differentiate between a manually-graded question (which may have a markdown solution and should therefore have all markdown cells in the solution block removed) and an autograded question. this means that instructors can have solutions to autograded questions which incorporate markup to help explain the answers or provide further information."

As a starting point, logic for stripping out the solution cells: https://github.com/ucbds-infra/otter-grader/blob/master/otter/assign/solutions.py#L199

### PR DESCRIPTION
**Pull request by aidansan**
Closes #730

### CODE REVIEW

File: otter/assign/solutions.py [hunk 1]
```diff
@@ -208,21 +208,46 @@ def strip_solutions_and_output(nb):
...
+    for i in del_md_solutions:
         del nb['cells'][i]
 
     # remove output from student version
     remove_output(nb)
 
     return nb
+
+SOLUTION_INCLUDE_MD_TAG = "otter_assign_solution_include_md"
+INCLUDE_MD_REGEX = re.compile(r"<!---?\s*\n*\s*INCLUDESTUDENT\s*\n*\s*-->\s*\n")
```
Comment (chrispyles) [inline otter/assign/solutions.py:234]: What is the purpose of requiring instructors to include `<!-- INCLUDESTUDENT -->` in the text of the markdown cell? Why not just have instructors add a tag to the cell like `otter_include` to the cell and just not remove the cell in the function above if that tag is present? That's the way other similar features are handled, and I'd prefer to use the same type of structure for this feature.

**Review by chrispyles — CHANGES_REQUESTED** [review]

**Comment by chrispyles** [comment]
Also please update the base of this PR to be the `beta` branch, not `master`, since this will be in v5.2.0, not a bugfix release for v5.1.

**Review by chrispyles — CHANGES_REQUESTED** [review]
Can you also add a test that this works correctly? You can update e.g. [this test notebook](https://github.com/ucbds-infra/otter-grader/blob/master/test/test_assign/files/example.ipynb) so that one of its questions uses this feature (and then update its [goldens](https://github.com/ucbds-infra/otter-grader/tree/master/test/test_assign/files/example-correct)). Instructions for running the test suite are [here](https://github.com/ucbds-infra/otter-grader/blob/master/CONTRIBUTING.md#running-tests) but the short version is to run tests for Otter Assign run `make test TESTPATH=test/test_assign`.

File: otter/assign/solutions.py (hunk 1 above)
Comment (aidansan) [reply otter/assign/solutions.py:234]: When you say add tags you mean like View -> Cell Toolbar ->  Tags and then manually add the tags?

Comment (chrispyles) [reply otter/assign/solutions.py]: Yes

**Comment by chrispyles** [comment]
Forgot to mention: can you also add an entry to the changelog for this change?

File: test/test_assign/files/example-correct/student/example.ipynb [hunk 2]
```diff
@@ -363,7 +426,7 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3 (ipykernel)",
```
Comment (aidansan) [inline test/test_assign/files/example-correct/student/example.ipynb:429]: Should I be concerned about this?

File: test/test_assign/files/example-correct/student/example.ipynb [hunk 3]
```diff
@@ -377,7 +440,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.7"
+   "version": "3.11.4"
```
Comment (aidansan) [inline test/test_assign/files/example-correct/student/example.ipynb:443]: Or this?

File: test/test_assign/files/example-correct/student/example.ipynb (hunk 2 above)
Comment (chrispyles) [reply test/test_assign/files/example-correct/student/example.ipynb:429]: it's fine

File: test/test_assign/files/example-correct/student/example.ipynb (hunk 3 above)
Comment (chrispyles) [reply test/test_assign/files/example-correct/student/example.ipynb:443]: ditto

**Review by chrispyles — APPROVED** [review]

File: test/test_assign/files/example-correct/student/example.ipynb [hunk 4]
```diff
@@ -315,6 +315,69 @@
...
+    "**Question 9:** Testing `otter_include` tag for markdown cells within a solution. Write a function `add`.\n",
+    "\n",
+    "_Points:_ 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_Type your answer here, replacing this text._"
```
Comment (aidansan) [inline test/test_assign/files/example-correct/student/example.ipynb:334]: This cell appearing is a bug right?

Comment (aidansan) [reply test/test_assign/files/example-correct/student/example.ipynb:334]: I think I need to update these lines to check for the tag? https://github.com/ucbds-infra/otter-grader/blob/e31415bc1c730b6704852c34756078633a225a0d/otter/assign/notebook_transformer.py#L293-L294

Comment (chrispyles) [reply test/test_assign/files/example-correct/student/example.ipynb:334]: Thanks for catching that. Yes you should just need to add `and question.manual` onto the condition in that if statement.

File: otter/assign/notebook_transformer.py [hunk 5]
```diff
@@ -203,7 +203,7 @@ def _get_transformed_cells(self, cells):
                     solution_has_md_cells, prompt_insertion_index = False, None
 
                 elif block_type is BlockType.SOLUTION:
-                    if not has_prompt and solution_has_md_cells:
+                    if not has_prompt and solution_has_md_cells and question.manual:
```
Comment (aidansan) [inline otter/assign/notebook_transformer.py:206]: I ended up modifying this if statement, does this seem correct?

Comment (chrispyles) [reply otter/assign/notebook_transformer.py:206]: LGTM

**Pull request merged**